src/javascript/app_2/pages/portfolio/portfolio.jsx

```diff
@@ -0,0 +1,277 @@
+import React from 'react';
+import moment from 'moment';
+import DAO from '../../data/dao';
+import DataTable from '../../components/elements/data_table.jsx';
+import Client from '../../../app/base/client';
+import Loading from '../../../../templates/_common/components/loading.jsx';
+import { buildOauthApps } from '../../../app/common/get_app_details';
+import { formatMoney } from '../../../app/common/currency';
+import { localize } from '../../../_common/localize';
+import { getPropertyValue } from '../../../_common/utility';
+
+const formatPortfolioData = (portfolio_arr) => {
+    const formatted_portfolio = portfolio_arr.map((portfolio_item) => {
+        const date_obj        = new Date(portfolio_item.expiry_time* 1000);
+        const moment_obj      = moment.utc(date_obj);
+        const remaining_time  = `${moment_obj.fromNow()} - ${moment_obj.format('h:mm:ss')}`;
+        const purchase        = parseFloat(portfolio_item.buy_price);
+        const payout          = parseFloat(portfolio_item.payout);
+
+        return {
+            ref       : portfolio_item.transaction_id,
+            type      : portfolio_item.contract_type,
+            details   : localize(portfolio_item.longcode.replace(/\n/g, '<br />')),
+            purchase  : formatMoney(false, purchase, true),
+            payout    : formatMoney(false, payout, true),
+            remaining_time,
+            id        : portfolio_item.contract_id,
+            app_id    : portfolio_item.app_id,
+            indicative: {
+                amount: '',
+                style : '',
+            },
+        };
+    });
+    return formatted_portfolio;
+};
+
+// TODO: move to common
+const contract_type_display = {
+    ASIANU      : {name: localize('Asian Up'), icon: 'asia_up'},
```

Comment:
How about using lowercase contract_types as filenames and put all in one folder according to theme or usage, then no need to map the `icon` everywhere?
(We need to use a different image for every single contract type. e.g. `CALL` and `CALLE` should be different)

Reply:
Yeah that is a good idea, could we perhaps use a sprite sheet for all the icons for binary static?